lib/statsd/instrument/prometheus/prometheus_sink.rb

```diff
@@ -57,15 +59,30 @@ def <<(datagram)
 
         private
 
+        def request_body(datagram)
+          aggregated = StatsD::Instrument::Prometheus::Aggregator.new(datagram).run
+          serialized = StatsD::Instrument::Prometheus::Serializer.new(aggregated).run
+          Snappy.deflate(serialized)
+        end
+
+        def make_request(datagram)
+          request = Net::HTTP::Post.new(uri.request_uri)
+          request['Authorization'] = "Bearer #{auth_key}"
+          request.body = request_body(datagram)
+          socket.request(request)
+        end
+
         def invalidate_socket
           socket = thread_store.delete(object_id)
-          socket&.close
+          socket&.finish
         end
 
         def socket
           thread_store[object_id] ||= begin
-            socket = UDPSocket.new
-            socket.connect(@host, @port)
+            socket = Net::HTTP.new(uri.host, uri.port)
+            socket.use_ssl = true
+            socket.set_debug_output($stdout) # TODO: remove
+            socket.start
```

Comment:
TODO: add timeouts. I think we can specify both `open_timout` and `read_timeout` to `start`.

Reply:
Added in c354a11